### DISCUSSION TITLE
.NET out-of-the-box/easier integration?

### DISCUSSION THREAD

**Discussion by connorjs** [microsoft/typespec#2831] [Ideas]
Hey there TypeSpec team and community 👋🏻

One hurdle I face with TypeSpec in a Microsoft/.NET (read: back-end first) enterprise environment is unfamiliarity with the npm/JS/TS ecosystem.

Have there been any discussions/are there any plans to provide out-of-the-box/easier* integration for .NET environments?

\*By "easier," I mean "simplifying CI configuration for a .NET API/service (ADO Pipelines or GitHub Actions) with respect to node/npm, running npm install/ci, and generating server stubs/swagger docs." Maybe this already is "simple enough" (UseNode, NpmCommand) and I just need to look at others’ examples.

Cheers

**Comment by timotheeguerin** [comment]
I do not believe there is any plans to make this particular workflow easier today. We do have a `msbuild` task that will compile the typespec as part of your build but this has been kept to the side for now for a specific azure usage. However this task only runs compile so you still need to manage the node installation and `npm install` yourself.

Curious of what you find today painful to setup in your CI environment. Would some documentation on how to compile TypeSpec with some common CI system help?
